internal/controller/resourcegroup/controller.go

```diff
@@ -80,6 +81,7 @@ func NewResourceGroupReconciler(
 func (r *ResourceGroupReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.ResourceGroup{}).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
```

Comment:
i've observed some weird infinite-loops during integration test logs... While i know that predicate event filters can fix it, i'm still slightly hesitant, it should be needed..

Need to do some careful sleuthing here